neuron.schema.json

```diff
@@ -0,0 +1,148 @@
+                        "type": "string"
+                    },
+                    "type": {
+                        "type": "string",
+                        "enum": ["scalar","vector"]
+                    }
+                },
+                "required": ["name", "type"],
+                "additionalProperties": false
+            }
+        },
+        "derivedParameters": {
+            "description": "All the names and derivation functions of secondary parameters in the simulation",
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string"
+                    },
+                    "derivation": {
+                        "type": "object",
+                        "minProperties": 1,
+                        "properties": {
+                            "generic": {"type": "string", "minlength": 1},
+                            "python": {"type": "string", "minlength": 1},
+                            "cpp": {"type": "string", "minlength": 1}
+                        },
+                        "additionalProperties": {"type": "string", "minlength": 1}
+                    }
+                },
+                "required": ["name", "derivation"],
+                "additionalProperties": false
+            }
+        },
+        "additionalParameters": {
+            "description": "The names and types of additional parameters for the simulation",
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string"
+                    },
+                    "type": {
+                        "type": "string",
+                        "enum": ["scalar","vector"]
+                    }
+                },
+                "required": ["name", "type"],
+                "additionalProperties": false
+            }
+        },
+        "additionalVariables": {
+            "description": "The names and types of additional variables for the simulation",
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string"
+                    },
+                    "type": {
+                        "type": "string",
+                        "enum": ["scalar","vector"]
+                    },
+                    "initial": {
+                        "type": "number"
+                    }
+                },
+                "required": ["name", "type", "initial"],
+                "additionalProperties": false
+            }
+        },
+        "simulationCode": {
+            "description": "The code for the simulation function of the neuron",
+            "type": "array",
+            "items": {
+                "type": "string"
```

Comment:
Do we want an array of strings for separate code lines or do we want one string (containing line breaks \n for readability)?

Reply:
I _thought_ an array would be better (_seems_ to be more standard too!) as we can actually have multiple lines in JSON that was (as far as I know using /n just leads to one long line in the source) but either will do... it's all programmatic anyway

Reply:
A (very) quick google suggests that this is a standard approach as it looks slightly prettier than having lots of \ns everywhere